File: Factory Method/Car.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Factory_Method
{
    internal class Car : IMotorVehicle
    {
        public void Build(string typeName)
        {
            Client.Instance.Invoke((MethodInvoker)(() => Client.Instance.ClearImage()));
            Client.Instance.Invoke((MethodInvoker)(() => Client.Instance.ShowMessage(typeName + " створюється...")));
            Task.Delay(2000).Wait();
            Client.Instance.Invoke((MethodInvoker)(() => Client.Instance.ShowMessageWithImage("Створено " + typeName, "car.jpg")));
        }
    }
}

File: Java/FactoryMethod/src/Car.java
class Car implements IMotorVehicle {
    public void build(String typeName) {
        Client.Instance.showMessage(typeName + " is being created...");
        try {
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        Client.Instance.showMessageWithImage("Created " + typeName, "car.jpg");
    }
}